### ISSUE TITLE
add support for parameter store

### ISSUE THREAD

**Issue by runningman84** [chef-cookbooks/aws#327]

**Comment by chefbob** [comment]
I started working on this but it is the first time I've written a Chef resource or really written anything in Ruby.  I've based my code on some of the other resources in this project.  My thought is to put the parameters into environment variables.  Does that make sense?  My code looks like this
```ruby
action :get do
    request = {
      name: name,
      with_decryption: with_decryption,
    }
    response = ssm_client.get_parameter(request)
    ENV["#{name}"]="#{response.parameter.value}"
    Chef::Log.debug "Get parameter #{name}"
end
```
If anyone wants to work on this with me please let me know.

**Comment by shatil** [comment]
@chefbob and @runningman84, how do you see the retrieved Parameter being used? For example, in Parameter Store, I keep some state configuration files to identify similar nodes, but I also keep secrets.

In the case of secrets, I'd want to use them in Chef templates, rather than writing them directly to files--which is what I'd do with the state config files, though.

**Comment by runningman84** [comment]
Yes I would also use them in templates to populate settings like db_user, db_password and so on.

I could also imagine to use it in the recipe, for example to use aws_s3_file with a s3 bucket name from a ssm value.

**Comment by chefbob** [comment]
I agree.  The idea would be to have them available to both templates and recipes.  I've spoken to a few people that the get should be done using a library instead of a resource.  I'm trying to figure out how to do that.

**Comment by chefbob** [comment]
create and delete were merged into master.  Going to work on get next and document it in the readme.

**Comment by cloneluke** [comment]
@chefbob  any ETA on get method?

**Comment by chefbob** [comment]
I have a pull request for issue 327.  Some of the commits don't have DCOs so I'm trying to fix that now.

**Comment by chefbob** [comment]
Even though I commit with -s, and my commit says signed off by chefbob <>, I keep getting the DCO error.  Not sure what I'm doing wrong?  Any suggestions?  The code I've written is here, https://github.com/chefbob/aws

**Comment by chefbob** [comment]
Finally got the DCO sorted out.  Pull request #337

**Comment by jeffbyrnes** [comment]
I think this can be closed, since #339 has merged in?
